riffdog/resources/ec2_instances.py

```diff
@@ -0,0 +1,132 @@
+"""
+This module is for EC2 instance processing - terraform & boto and comparison.
+"""
+
+import logging
+
+from ..utils import _get_client, _get_resource
```

Comment:
Any reason why these can't be on the class?

Edit: or on the base `Resource` class since the are shared

Reply:
So, Thats valid - but what if you didn't wanat to make an AWS resource in the future.. perhaps there should be an AWS Resource class that implements resource that has these, and then future people could make (for example) a Cloudflare Class that has the get__cloudflare_client (or whatever)?

Reply:
I like the idea of the Resource hierarchy for future extensibility

Reply:
This is implemented, but might want to think about the folder namespacing of the `resource` module (not the `resources` which we are talking about separately, which also will need doing)

Reply:
Cool, looks good.